usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -287,6 +278,30 @@ def query_elasticsearch(self) -> list:
         filter_query = QueryWithFilters.generate_awards_elasticsearch_query(self.filters)
         sort_field = self.get_elastic_sort_by_fields()
         sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
+        record_num = (self.pagination["page"] - 1) * self.pagination["limit"]
+        if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and not self.last_id and not self.last_value:
```

Comment:
Are `self.last_id` and `self.last_value` taken as integers or strings? And can one be populated and not the other? 

Questions pointing towards some ambiguity. It may be best to use `None` as a comparison rather than falsey in case `total_obligation` is valid as `0`. 

Also adding parentheses: 

If both have to be there to to be used in manually specifying what page to land on:
`if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and (self.last_id is None and self.last_value is None):`

If only one needs to be there to be used in manually specifying what page to land on:
`if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and not (self.last_id is None and self.last_value is None):`

... though I think I may have messed up that last one, but hopefully you get the gist.

Reply:
last_id should be an integer, last_value can be any of the sort fields, so it should be a string. The none comparison is a good point, will change